File: ruby/src/Behavioral/Template_Method/BookTitleInfo.rb

#//BookTitleInfo - two of three concrete templates

require "Behavioral/Template_Method/TitleInfo"

class BookTitleInfo < TitleInfo

	def initialize(titleName,author)
 		setTitleName(titleName)
		setAuthor(author)
	end

	def setAuthor (a)
 		@author = a
	end

	def getAuthor ()
 		return @author
	end

	def getTitleBlurb ()
		"Book: #{getTitleName()}, Author: #{getAuthor() }"
	end
end

File: csharp/src/Behavioral/Template_Method/BookTitleInfo.cs
namespace Behavioral.Template_Method {


//BookTitleInfo.java - two of three concrete templates

public class BookTitleInfo : TitleInfo {  
   private string author;
    
   public BookTitleInfo(string titleName, string author) {
       this.setTitleName(titleName);
       this.setAuthor(author);
   }    
   
   public void setAuthor(string authorIn) {this.author = authorIn;}
   public string getAuthor() {return this.author;}   
   
   override public string getTitleBlurb() {
       return ("Book: " + this.getTitleName() + 
               ", Author: " + this.getAuthor());
   }
}

}
